api/src/backend/api/v1/views.py

```diff
@@ -2565,3 +2572,126 @@ def get_serializer_context(self):
         context = super().get_serializer_context()
         context["allowed_providers"] = self.allowed_providers
         return context
+
+
+class LighthouseConfigViewSet(BaseRLSViewSet):
```

Comment:
The CRUD endpoints could be omitted to avoid undesired behavior if they do not need to perform any extra action. If they just manage the resource in the database, it's not necessary to define the functions and the view is left clearer.

Reply:
I've removed `update` and `destroy` as they are not really doing anything. For create we need the custom `LighthouseConfigSerializer` serializer to have a masked key in output. Update also used the serializer but the response doesn't contain the API key field and that's okay.